stylus-tools/src/cargo_stylus.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2025, Offchain Labs, Inc.
+// For licensing, see https://github.com/OffchainLabs/stylus-sdk-rs/blob/main/licenses/COPYRIGHT.md
+
+use alloy::primitives::Address;
+use eyre::{bail, eyre, Result, WrapErr};
+use regex::Regex;
+use std::process::Command;
+
+/// Deploy the contract in the current directory
+pub fn deploy(rpc: &str, key: &str) -> Result<Address> {
```

Comment:
Eventually, this will be replaced by a library function that doesn't need to call the cargo-stylus binary. I didn't want to do that now because I don't want to cause conflicts with @rory-ocl 's work.